File: src/com/castlefrog/agl/IllegalActionException.java
package com.castlefrog.agl;

public class IllegalActionException extends IllegalArgumentException {
    private static final long serialVersionUID = 1L;

    public <S extends State, A extends Action> IllegalActionException(A action, S state) {
        super("Illegal action, " + action + ", from state," + state);
    }
}

File: src/main/kotlin/com/castlefrog/agl/IllegalActionException.kt
package com.castlefrog.agl

/**
 * Thrown by [Simulator.stateTransition] when the actions list is invalid for the
 * given state: wrong length, missing action for the player to move, null when an
 * action is required, or an action that is not legal.
 *
 * Callers implementing generic agent/simulator loops can catch this type alone
 * for “illegal transition” errors across all domains.
 */
class IllegalActionException(message: String, cause: Throwable? = null) :
    IllegalArgumentException(message, cause)
